### PR TITLE
Define `Expandable` type

### PR DESCRIPTION
**Pull request by kestred**
Begin implementing https://github.com/wyyerd/stripe-rs/issues/56.

@stearnsc - I've re-jiggered the `Identifiable` that you wrote, so would definitely love your thoughts if you've got time.

### CODE REVIEW

**Comment by stearnsc** [comment]
I might rename `Paginated` to `Paginate` (something that can be paginated) but otherwise lgtm (for the record, I think I should have called `Identifiable` `Identify` for consistency :p)

File: src/params.rs [hunk 1]
```diff
@@ -10,12 +10,98 @@ pub struct Headers {
...
+            Expandable::Id(_) => None,
+            Expandable::Object(obj) => Some(*obj),
+        }
+    }
+}
+
+/// Implemented by types which support cursor-based pagination,
+/// typically with an id, allowing them to be fetched using a `List`
+/// returned by the corresponding "list" api request.
+pub trait Paginated {
```
Comment (stearnsc) [inline src/params.rs]: Maybe rename to `Paginate`?

File: src/resources/invoice.rs [hunk 2]
```diff
@@ -115,57 +116,72 @@ pub struct InvoiceLineItem {
...
-    pub subscription_proration_date: Option<Timestamp>,
     pub subtotal: i64,
     pub tax: Option<i64>,
     pub tax_percent: Option<f64>,
     pub total: i64,
     pub webhooks_delivered_at: Option<Timestamp>,
 }
 
+impl Object for Invoice {
+    type Id = Option<InvoiceId>; // an invoice id can be none if it is an _upcoming_ invoice
```
Comment (stearnsc) [inline src/resources/invoice.rs]: bleh. Does this mean `Invoice` can't be paginated?

Comment (kestred) [reply src/resources/invoice.rs]: It works the same as before, Invoice isn't `Paginated` but there is the same custom `List` impl below

**Pull request merged**